src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -853,7 +853,7 @@ private void ProcessScanlineFromPalette<TPixel>(ReadOnlySpan<byte> defilteredSca
             where TPixel : struct, IPixel<TPixel>
         {
             ReadOnlySpan<byte> newScanline = ToArrayByBitsLength(defilteredScanline, this.bytesPerScanline, this.header.BitDepth);
-            byte[] pal = this.palette;
+            ReadOnlySpan<Rgb24> pal = MemoryMarshal.Cast<byte, Rgb24>(this.palette);
```

Comment:
won't this fail if `this.palette` isn't the correct length? just raising this as a subtle change of behavior that might introduce exceptions that previously (potentially incorrectly) allowed.

**note:** not asking for a change just pointing it out incase someone else thinks its important.

Reply:
If we didn't before, we should throw if the image tries to index into the palette outside it's bounds.